### PR TITLE
topN: define behavior for nth >= r.length, improve speed

### PR DESCRIPTION
**Pull request by andralex**
* There was an unlisted bug - if `nth > r.length` the doc says "has no effect" but what happened was a range violation.
* Eliminate a redundant test
* Add special casing for the "max" case. Saves a whole lot of work and could happen during the algorithm if the user specifies a large-ish `nth` (e.g. median).

### CODE REVIEW

**Comment by andralex** [comment]
@9il could you see whether this improves your test cases?

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -2185,14 +2185,17 @@ auto topN(alias less = "a < b",
...
 {
-    import std.algorithm : swap; // FIXME
-    import std.random : uniform;
-
     static assert(ss == SwapStrategy.unstable,
             "Stable topN not yet implemented");
+
+    if (nth >= r.length) return r[0 .. $];
+
+    import std.algorithm : swap; // FIXME
```
Comment (quickfur) [inline std/algorithm/sorting.d]: Why "FIXME"? Isn't this how local imports are supposed to work?

Oh I see, it's inherited from older code... looks like the intention was to replace it with `import std.algorithm.mutation : swap;`.

File: std/algorithm/sorting.d [hunk 2]
```diff
@@ -2200,6 +2203,13 @@ auto topN(alias less = "a < b",
             swap(r.front, r.minPos!less.front);
             break;
         }
+        if (nth + 1 == r.length)
+        {
+            // Special-case "max"
+            import std.algorithm.searching : minPos;
```
Comment (quickfur) [inline std/algorithm/sorting.d]: Shouldn't this be merged with the same import in the previous if-block and hoisted one lexical scope up?

File: std/algorithm/sorting.d (hunk 1 above)
Comment (andralex) [inline std/algorithm/sorting.d]: It was there from a while ago. I'll remove it.

File: std/algorithm/sorting.d (hunk 2 above)
Comment (andralex) [inline std/algorithm/sorting.d]: Meh, ok.

**Comment by andralex** [comment]
@klickverbot fair enough, added

**Comment by 9il** [comment]
`topN` with this PR is 5-7% faster comparing with git master.

**Pull request merged**